Add linting

We don't have any CI process setup against this repository; add it.

At the least we're seeing failures now:

```
frodo ~/Repos/github.com/skx/tcl $ ~/go/lint 
interpreter/interpreter.go:310:10: if block ends with a return statement, so drop this else and outdent its block
interpreter/interpreter.go:316:10: if block ends with a return statement, so drop this else and outdent its block
interpreter/interpreter.go:322:10: if block ends with a return statement, so drop this else and outdent its block
interpreter/interpreter.go:328:10: if block ends with a return statement, so drop this else and outdent its block
interpreter/interpreter.go:369:9: if block ends with a return statement, so drop this else and outdent its block
interpreter/interpreter.go:469:3: this value of err is never used (SA4006)
lexer/lexer.go:29:2: field decimal is unused (U1000)
frodo ~/Repos/github.com/skx/tcl $ 

```

--- a/.github/FUNDING.yml
+++ b/.github/FUNDING.yml
@@ -1,0 +1,3 @@
+# These are supported funding model platforms
+github: skx
+custom: https://steve.fi/donate/

--- a/.github/run-tests.sh
+++ b/.github/run-tests.sh
@@ -1,0 +1,4 @@
+#!/bin/bash
+
+# Run our golang tests
+go test ./... -race

--- a/.github/workflows/pull_request.yml
+++ b/.github/workflows/pull_request.yml
@@ -1,0 +1,18 @@
+on: pull_request
+name: Pull Request
+jobs:
+  golangci:
+    name: lint
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/setup-go@v2
+      - uses: actions/checkout@v2
+      - name: golangci-lint
+        uses: golangci/golangci-lint-action@v2
+  test:
+    name: Test
+    runs-on: ubuntu-latest
+    steps:
+    - uses: actions/checkout@master
+    - name: Test
+      uses: skx/github-action-tester@master

--- a/.github/workflows/push.yml
+++ b/.github/workflows/push.yml
@@ -1,0 +1,13 @@
+on:
+  push:
+    branches:
+    - master
+name: Push Event
+jobs:
+  test:
+    name: Run tests
+    runs-on: ubuntu-latest
+    steps:
+    - uses: actions/checkout@master
+    - name: Test
+      uses: skx/github-action-tester@master

--- a/interpreter/interpreter.go
+++ b/interpreter/interpreter.go
@@ -307,27 +307,30 @@ func expr(i *Interpreter, args []string) (string, error) {
 	case "<":
 		if aV < bV {
 			return "1", nil
-		} else {
-			return "0", nil
 		}
+		return "0", nil
+	case "==":
+		if aV == bV {
+			return "1", nil
+		}
+		return "0", nil
 	case "<=":
 		if aV <= bV {
 			return "1", nil
-		} else {
-			return "0", nil
 		}
+		return "0", nil
+
 	case ">":
 		if aV > bV {
 			return "1", nil
-		} else {
-			return "0", nil
 		}
+		return "0", nil
+
 	case ">=":
 		if aV >= bV {
 			return "1", nil
-		} else {
-			return "0", nil
 		}
+		return "0", nil
 	}
 	return "", fmt.Errorf("unknown operation %s %s %s", args[0], op, args[2])
 }
@@ -366,11 +369,11 @@ func iff(i *Interpreter, args []string) (string, error) {
 	// A non-false result means we run the "true branch"
 	if out != "" && out != "0" {
 		return i.Eval(pass)
-	} else {
-		// If we have a "false branch", then execute it
-		if fail != "" {
-			return i.Eval(fail)
-		}
+	}
+
+	// If we have a "false branch", then execute it
+	if fail != "" {
+		return i.Eval(fail)
 	}
 
 	return "", nil
@@ -467,8 +470,11 @@ func while(i *Interpreter, args []string) (string, error) {
 
 		// run the body
 		out, err = i.Eval(body)
+		if err != nil {
+			return "", err
+		}
 
-		// repeat the
+		// repeat the conditional-test ahead of repeating the body
 		tmp, err = i.Eval(cond)
 		if err != nil {
 			return "", err

--- a/lexer/lexer.go
+++ b/lexer/lexer.go
@@ -334,7 +334,6 @@ func (l *Lexer) readNestedPair(open rune, close rune) (string, error) {
 
 	}
 
-	return out, nil
 }
 
 // readVariable returns a variable

--- a/lexer/lexer.go
+++ b/lexer/lexer.go
@@ -26,7 +26,6 @@ import (
 // Lexer is used as the lexer for our deployr "language".
 type Lexer struct {
 	debug        bool                 // dump tokens as they're read?
-	decimal      bool                 // convert numbers to decimal?
 	position     int                  // current character position
 	readPosition int                  // next character position
 	ch           rune                 // current character

